Tropos/Controllers/TRWeatherController.m

```diff
@@ -62,15 +59,30 @@ - (instancetype)init
     return self;
 }
 
+- (RACSignal *)weatherUpdateChanged
+{
+    TRWeatherUpdate *cachedUpdate = [[TRWeatherUpdateCache new] latestWeatherUpdate];
+    RACSignal *weatherUpdates = [self.updateWeatherCommand.executionSignals startWith:[RACSignal return:cachedUpdate]];
+
+    return [[[[weatherUpdates switchToLatest] filter:^BOOL(TRWeatherUpdate *update) {
+        return update != nil;
+    }] doNext:^(TRWeatherUpdate *update) {
+        [[TRAnalyticsController sharedController] trackEvent:update];
+        [[TRWeatherUpdateCache new] archiveWeatherUpdate:update];
```

Comment:
I think that makes sense. How about https://github.com/thoughtbot/Tropos/commit/8f82761ebc8fc0f3a95ec9af43bcbbbf0c1b38a7 ?